### ISSUE TITLE
Inconsistent width of PathArcTo strokes

### ISSUE THREAD

**Issue by pixtur** [ocornut/imgui#4993]
**Version/Branch of Dear ImGui:**

Version: 1.83
Branch: master

**Back-end/Renderer/Compiler/OS**

Back-ends imgui .net
Operating System: windows 10

**My Issue/Question:**
After upgrading from 1.77 to 1.86 there seem to appear rendering glitches for paths that include PathArcTo() with a small radius.

It's not unlikely that I'm not using the API correctly. But the effect only appeared recently and might be a regression of some optimization.

**Screenshots/Video**
As screenshot highlighting some issues.
![Frame 23](https://user-images.githubusercontent.com/1732545/152853229-212160ef-a788-4c3c-bc75-28bd98f39855.png)

Here is a capture of the code below...
https://user-images.githubusercontent.com/1732545/152852430-57253e47-46aa-4239-bcd4-939993eb1bef.mp4

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_

Sorry. c# only... 🙈

```
        private void DrawTest()
        {
            if (ImGui.Begin("test"))
            {
                ImGui.DragFloat("radius", ref _testRadius, 0.1f);
                ImGui.DragFloat("thickness", ref _thickness, 1f);
                
                const float Pi = 3.141592f;
                var drawList = ImGui.GetWindowDrawList();

                var center = new Vector2(300, 400);
                var max = 100;
                
                for (float i = _testRadius; i < max; i += 20)
                {
                    var radius = 10 + i;
                    drawList.PathClear();
                    drawList.PathArcTo(center + new Vector2(0, -100), max- radius,  1.5f * Pi,Pi );
                    drawList.PathArcTo(center + new Vector2(-max,10), radius, 0, Pi * 0.5f);
                    drawList.PathLineTo( center + new Vector2(-max - 100,10 + radius) );
                    
                    drawList.AddPolyline(ref drawList._Path[0], drawList._Path.Size, Color.White, ImDrawFlags.None, _thickness + 2);
                    drawList.PathStroke(Color.DarkGray, ImDrawFlags.None, _thickness);
                }

                ImGui.End();
            }
        }
```

**Comment by thedmd** [comment]
I got it fixed in [775b9e4](https://github.com/thedmd/imgui/commit/775b9e4939ab6ae4717ae549b9de50d0e005d22a).

![image](https://user-images.githubusercontent.com/1197433/152988162-c80a1a00-549a-4bb8-9fd2-7a124bfc506d.png)

**Comment by pixtur** [comment]
Wow! That's incredible.

One more thing... I noticed that ImGui.net v1.84 can cause division/0 exceptions for `PathArcTo()` and `DrawCicle()` when called with  small radii (e.g. 1e-7).  I'm not sure if this is related do your fix. But these exceptions seem to be introduced the same time, as the drawing glitches.

**Comment by ocornut** [comment]
Merged @thedmd fix as 4691fa0

Thank you very much !
